src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -166,9 +177,63 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
             GenerateRequestBuilderCodeFiles(modelsNamespace);
             GroupReusableModelsInSingleFile(modelsNamespace);
             RemoveSelfReferencingUsings(generatedCode);
+            AddAliasToCodeFileUsings(generatedCode);
+            CorrectSerializerParameters(generatedCode);
             cancellationToken.ThrowIfCancellationRequested();
         }, cancellationToken);
     }
+
+    private static void CorrectSerializerParameters(CodeElement currentElement)
+    {
+        if (currentElement is CodeFunction currentFunction &&
+            currentFunction.OriginalLocalMethod.Kind is CodeMethodKind.Serializer)
+        {
+            foreach (var parameter in currentFunction.OriginalLocalMethod.Parameters
+                         .Where(p => GetOriginalComposedType(p.Type) is CodeComposedTypeBase composedType &&
+                                     composedType.IsComposedOfObjectsAndPrimitives(IsPrimitiveType)))
+            {
+                var composedType = (CodeComposedTypeBase)GetOriginalComposedType(parameter.Type)!;
```

Comment:
```suggestion
                var composedType = GetOriginalComposedType(parameter.Type)!;
```
also as the function `GetOriginalComposedType` can explicitly return null, we probably shouldn't supress the null warning but deal with it...